build/source/dshare/get_ixname.f90

```diff
@@ -836,9 +818,158 @@ function get_ixbvar(varName)
+  case('unknown'); get_ixVarType = iLookVarType%unknown
+  ! get to here if cannot find the variable
+  case default
+   get_ixVarType = integerMissing
+ endselect
+ end function get_ixVarType
+
+ ! ****************************************************************************************************************
+ ! public function get_varTypeName: get the index of the named variable type
+ ! ****************************************************************************************************************
+ function get_varTypeName(varType)
+ USE var_lookup,only:iLookVarType                    ! indices of the named variable types
+ implicit none
+ ! define dummy variables
+ integer(i4b), intent(in) :: varType                 ! variable type name
+ character(LEN=7)         :: get_varTypeName         ! index of the named variable type list
+ ! get the index of the named variables
+ select case(varType)
+  case(iLookVarType%scalarv);get_varTypeName='scalarv'
+  case(iLookVarType%wLength);get_varTypeName='wLength'
+  case(iLookVarType%midSnow);get_varTypeName='midSnow'
+  case(iLookVarType%midSoil);get_varTypeName='midSoil'
+  case(iLookVarType%midToto);get_varTypeName='midToto'
+  case(iLookVarType%ifcSnow);get_varTypeName='ifcSnow'
+  case(iLookVarType%ifcSoil);get_varTypeName='ifcSoil'
+  case(iLookVarType%ifcToto);get_varTypeName='ifcToto'
+  case(iLookVarType%routing);get_varTypeName='routing'
+  case(iLookVarType%unknown);get_varTypeName='unknown'
+  ! get to here if cannot find the variable
+  case default
+   get_VarTypeName = 'missing'
+ endselect
+ end function get_VarTypeName
+
+ ! *******************************************************************************************************************
+ ! public subroutine get_ixUnknown: get the index of the named variable type from ANY structure, as well as the 
+ ! structrue that it was found in
+ ! *******************************************************************************************************************
+ subroutine get_ixUnknown(varName,typeName,vDex,err,message)
+ USE nrtype
+ USE globalData,only:structInfo        ! information on the data structures                  
+ USE multiconst,only:integerMissing    ! missing integer value
+ implicit none
+
+ ! dummies
+ character(*),intent(in)  :: varName   ! variable name
+ character(*),intent(out) :: typeName  ! variable type name
+ integer(i4b),intent(out) :: vDex      ! variable index in structure
+ integer(i4b),intent(out) :: err       ! error code
+ character(*),intent(out) :: message   ! error message
+
+ ! internals
+ integer(i4b)             :: iStruc    ! index for looping through structure types
+
+ ! error init
+ err=0
+ message='get_ixUnknown/'
+
+ ! loop through all structure types to find the one with the given variable name
+ ! pill variable index plus return which structure it was found in
+ do iStruc = 1,size(structInfo)
+  select case(trim(structInfo(iStruc)%structName))
```

Comment:
It would be nice to make this routine easier to read. For example,
```
do iStruc = 1,size(structInfo)
 select case(trim(structInfo(iStruc)%structName))
  case('time'); vDex = get_ixTime(trim(varName))
  case('forc'); vDex = get_ixForce(trim(varName))
  ..
 end select
 if(vDex>0)then
  typeName=trim(structInfo(iStruc)%structName)
  return
 endif
end do
```

Note that the if statement is only called once.

Also, this code is quite similar to some of the code in `checkStruc.f90` -- it would be good to merge if possible.

In `checkStruc.f90` I also had an error if a variable is in multiple data structures. This can be done as

```
vDexSave = integerMissing
do iStruc = 1,size(structInfo)
 select case(trim(structInfo(iStruc)%structName))
  case('time'); vDex = get_ixTime(trim(varName))
  case('forc'); vDex = get_ixForce(trim(varName))
  ..
  case default
   message=trim(message)//'unable to identify variable'
   err=20; return
 end select
 if(vDex>0)then
  if(vDexSave>0)then
   message=trim(message)//'variable '//trim(varname)//'in multiple structures'
   err=20; return
  endif
  typeName=trim(structInfo(iStruc)%structName)
 endif
end do
```